woocommerce/woocommerce
# Randomize the Sections in Recommended Marketing Extensions

### All Submissions:






### Changes proposed in this Pull Request:



Closes #33657.

This randomizes the **section** placement in the Recommended Marketing Extensions to ensure extensions are distributed evenly "above the fold". At present, this means the "Grow your store" and "Reach out to customers" sections now switch placements randomly.

<img width="709" alt="Screen Shot 2022-07-13 at 8 28 04 pm" src="https://user-images.githubusercontent.com/9312929/178734168-02c20631-87d5-4fd7-a51d-c49262d4d4c0.png">

A new `section_order` prop is also added to the Tracks event `tasklist_marketing_install`. This should contain a comma separated list of the sections at the time of install. 

### How to test the changes in this Pull Request:

1. On a fresh site, click the Marketing Task (Get more sales).
2. Refresh the Marketing Task page a few times. The "Grow your store" and "Reach out to customers" sections should randomly switch placements.

#### Tracks
1. Install one of the extensions using the "Get started" button and confirm the `section_order` prop in `tasklist_marketing_install` matches the section order currently listed on screen.

### Other information:




### FOR PR REVIEWER ONLY:
